File: backend/src/main/java/ru/fbtw/thestore/backend/data/order/dto/MyOrderDto.java
package ru.fbtw.thestore.backend.data.order.dto;

import lombok.*;
import ru.fbtw.thestore.backend.data.catalog.dto.CompactProductDto;
import ru.fbtw.thestore.backend.data.delivery.dto.DeliveryDto;
import ru.fbtw.thestore.backend.data.delivery.dto.ShopDto;
import ru.fbtw.thestore.backend.data.payment.dto.PaymentDto;

import java.io.Serializable;
import java.util.Set;

/**
 * A DTO for the {@link ru.fbtw.thestore.backend.domains.order.MyOrder} entity
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MyOrderDto implements Serializable {
	private Long id;
	private ShopDto shop;
	private String orderTotal;
	private boolean orderStatus;
	private DeliveryDto delivery;
	private PaymentDto payment;
	private Set<CompactProductDto> products;
}

File: mobile_app/the_store_app/lib/entity/order/my_order_dto.dart

import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:the_store_app/data/catalog/models/short_product.dart';
import 'package:the_store_app/entity/catalog/catalog_export.dart';
import 'package:the_store_app/entity/delivery/delivery_dto.dart';
import 'package:the_store_app/entity/payment/payment_dto.dart';
import 'package:the_store_app/entity/shop/shop_dto.dart';
part 'my_order_dto.freezed.dart';
part 'my_order_dto.g.dart';

@Freezed(
  fromJson: true,
  toJson: true,
)
class MyOrderDto with _$MyOrderDto{
  factory MyOrderDto({
    required int id,
    ShopDto? shop,
    required String orderTotal,
    required bool orderStatus,
    DeliveryDto? delivery,
    required PaymentDto paymentDTO,

    required List<ShortProduct> products,
}) = _MyOrderDto;

  factory MyOrderDto.fromJson(Map<String, dynamic> json) => _$MyOrderDtoFromJson(json);
}